packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1422,7 +1422,11 @@ abstract class ResidentRunner {
     }
     vmService.onExtensionEvent.listen((vm_service.Event event) {
       if (event.json['extensionKind'] == 'Flutter.FrameworkInitialization') {
-        completer.complete();
+        // The 'Flutter.FrameworkInitialization' event is sent on hot restart
```

Comment:
I would also cancel this subscription after completing

Reply:
done